app/screens/edit_profile/edit_profile.tsx

```diff
@@ -0,0 +1,399 @@
+        setUpdating(false);
+        enableSaveButton(true);
+        scrollViewRef.current?.scrollToPosition(0, 0, true);
+    };
+
+    const updateField = useCallback((fieldKey: string, name: string) => {
+        const update = {...userInfo};
+        update[fieldKey] = name;
+        setUserInfo(update);
+
+        // @ts-expect-error access object property by string key
+        const currentValue = currentUser[fieldKey];
+        const didChange = currentValue !== name;
+        enableSaveButton(didChange);
+    }, [userInfo]);
+
+    const includesSsoService = (sso: string) => ['gitlab', 'google', 'office365'].includes(sso);
+    const isSAMLOrLDAP = (protocol: string) => ['ldap', 'saml'].includes(protocol);
+
+    const fieldSequence: FieldSequence = {
+        firstName: {
+            ref: firstNameRef,
+            isDisabled: (isSAMLOrLDAP(service) && lockedFirstName) || includesSsoService(service),
+        },
+        lastName: {
+            ref: lastNameRef,
+            isDisabled: (isSAMLOrLDAP(service) && lockedLastName) || includesSsoService(service),
+        },
+        username: {
+            ref: usernameRef,
+            isDisabled: service !== '',
+        },
+        email: {
+            ref: emailRef,
+            isDisabled: true,
+        },
+        nickname: {
+            ref: nicknameRef,
+            isDisabled: isSAMLOrLDAP(service) && lockedNickname,
+        },
+        position: {
+            ref: positionRef,
+            isDisabled: isSAMLOrLDAP(service) && lockedPosition,
+        },
+    };
+
+    const onFocusNextField = (fieldKey: string) => {
+        const findNextField = () => {
+            const fields = Object.keys(fieldSequence);
+            const curIndex = fields.indexOf(fieldKey);
+            const searchIndex = curIndex + 1;
+
+            if (curIndex === -1 || searchIndex > fields.length) {
+                return undefined;
+            }
+
+            const remainingFields = fields.slice(searchIndex);
+
+            const nextFieldIndex = remainingFields.findIndex((f: string) => {
+                const field = fieldSequence[f];
+                return !field.isDisabled;
+            });
+
+            if (nextFieldIndex === -1) {
+                return {isLastEnabledField: true, nextField: undefined};
+            }
+
+            const fieldName = remainingFields[nextFieldIndex];
+            return {isLastEnabledField: false, nextField: fieldSequence[fieldName]};
+        };
+
+        const next = findNextField();
+        if (next?.isLastEnabledField && canSave) {
+            // performs form submission
+            Keyboard.dismiss();
+            submitUser();
+        } else if (next?.nextField) {
+            next?.nextField?.ref?.current?.focus();
+        } else {
+            Keyboard.dismiss();
+        }
+    };
+
+    const fieldConfig: FieldConfig = {
+        blurOnSubmit: false,
+        enablesReturnKeyAutomatically: true,
+        onFocusNextField,
```

Comment:
onFocusNextField rerenders every time, rerendering the children components. Is this expected?

Reply:
100% is re-rendering as is always a new object reference. this needs to be addressed

Reply:
Corrected